### PR TITLE
[WIP] Feature #3417 - Implement tuiDropdownDirectionChange & tuiHintDirectionChange

### LINKED ISSUE #3417: 🚀 - tuiDropdownDirection and tuiHintDirection Output
### Which @taiga-ui/* package(s) are relevant/releated to the feature request?

core, cdk

### Description

The drop down's direction is changed depending on available screen space, and this does not seem to be communicated to outside components. I would like to add the arrow styling from regular [tooltips](https://taiga-ui.dev/components/tooltip) but with the more powerful API of hosted dropdowns, and also change my template depending on whether it is being displayed on top or on the bottom. This would be possible if the calculated position were to be communicated to the outside.

### PR DESCRIPTION
**Pull request by shiv9604**
Close #3417

### CODE REVIEW

File: projects/core/directives/hint/hint.directive.ts [hunk 1]
```diff
@@ -80,4 +100,12 @@ export class TuiHintDirective<C>
             this.service.remove(this);
         }
     }
+
+    public hintDirectionObserver(): void {
+        this.service.hintDirection$
```
[the remark this answers is not in the document]
Comment (shiv9604) [reply projects/core/directives/hint/hint.directive.ts]: Thanks for your valuable feedback and clarification! Please check the resolved PR comments and let me know if anything is required.

**Review by waterplea — CHANGES_REQUESTED** [review]

File: projects/core/directives/hint/hint.directive.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (waterplea) [reply projects/core/directives/hint/hint.directive.ts]: Great! Added a few more comments, after they are addressed this looks good to me, thank you for your contribution!

File: projects/core/directives/dropdown/dropdown-position-sided.directive.ts [hunk 2]
```diff
@@ -57,11 +61,22 @@ export class TuiDropdownPositionSided extends TuiPositionAccessor {
...
             return [position[this.previous], left];
         }
 
         this.previous = better;
+        this.emitDirection(better);
 
         return [position[better], left];
     }
+
+    public emitDirection(direction: TuiVerticalDirection): void {
```
[the remark this answers is not in the document]
Comment (shiv9604) [reply projects/core/directives/dropdown/dropdown-position-sided.directive.ts]: Resolved

File: projects/core/directives/dropdown/dropdown.directive.ts [hunk 3]
```diff
@@ -34,7 +34,13 @@ import {TuiDropdownPosition} from './dropdown-position.directive';
         tuiAsVehicle(TuiDropdownDirective),
     ],
     exportAs: 'tuiDropdown',
-    hostDirectives: [TuiDropdownDriverDirective, TuiDropdownPosition],
+    hostDirectives: [
+        TuiDropdownDriverDirective,
+        {
+            directive: TuiDropdownPosition,
```
[the remark this answers is not in the document]
Comment (shiv9604) [reply projects/core/directives/dropdown/dropdown.directive.ts:40]: Resolved

File: projects/core/directives/dropdown/dropdown-position.directive.ts [hunk 4]
```diff
@@ -20,6 +21,10 @@ export class TuiDropdownPosition extends TuiPositionAccessor {
     private readonly viewport = inject(TUI_VIEWPORT);
 
     private previous?: TuiVerticalDirection;
+    private lastDirection!: TuiVerticalDirection;
```
[the remark this answers is not in the document]
Comment (shiv9604) [reply projects/core/directives/dropdown/dropdown-position.directive.ts]: Resolved

File: projects/core/directives/dropdown/dropdown-position-sided.directive.ts [hunk 5]
```diff
@@ -57,10 +56,13 @@ export class TuiDropdownPositionSided extends TuiPositionAccessor {
             (available[this.previous] > minHeight && direction) ||
             this.previous === better
         ) {
+            this.vertical.emitDirection(better);
```
Comment (waterplea) [inline projects/core/directives/dropdown/dropdown-position-sided.directive.ts:59]: It should be `this.previous` here

File: projects/core/directives/dropdown/dropdown-position.directive.ts [hunk 6]
```diff
@@ -60,7 +68,10 @@ export class TuiDropdownPosition extends TuiPositionAccessor {
                     : right,
             left: Math.max(viewport.left, left),
         } as const;
-        const better = available.top > available.bottom ? 'top' : 'bottom';
+        const better: TuiVerticalDirection =
+            available.top > available.bottom ? 'top' : 'bottom';
+
+        this.emitDirection(better);
```
Comment (waterplea) [inline projects/core/directives/dropdown/dropdown-position.directive.ts:74]: It's not always `better`, sometimes it's `previous`, see `if` on the next line.

File: projects/legacy/components/input-tag/input-tag.style.less [hunk 7]
```diff
@@ -285,10 +285,7 @@
 }
 
 @media @tui-mouse {
-    :host:not(._readonly):not(._disabled):hover
-        .t-textfield-icon
-        ::ng-deep
-        [tuiAppearance][data-appearance='icon']::after {
+    :host:not(._readonly):not(._disabled):hover .t-textfield-icon ::ng-deep [tuiAppearance][data-appearance='icon']::after {
```
Comment (waterplea) [inline projects/legacy/components/input-tag/input-tag.style.less:288]: Weird that this got affected 🤔 You should rollback those style changes. Maybe you need to run `npm i` to update some lint config packages locally?

File: projects/core/directives/dropdown/dropdown-position-sided.directive.ts [hunk 8]
```diff
@@ -10,13 +10,12 @@ import {TuiDropdownPosition} from './dropdown-position.directive';
 @Directive({
     standalone: true,
     selector: '[tuiDropdownSided]',
-    providers: [TuiDropdownPosition, tuiAsPositionAccessor(TuiDropdownPositionSided)],
+    providers: [tuiAsPositionAccessor(TuiDropdownPositionSided)],
```
Comment (waterplea) [inline projects/core/directives/dropdown/dropdown-position-sided.directive.ts:13]: Looks like this was still necessary. Please put it back — I'll investigate myself later how so.

**Pull request closed without merging**